### PR TITLE
Moody no longer affects accuracy or evasion.

### LINKED ISSUE #6027: Moody shouldn't change accuracy or evasion
What happens:
The moody ability boosts any stat by 2 at the end of each turn, then reduces another stat by 1.

What should happen:
The accuracy and evasion stats should not be boosted nor reduced.

Source from researcher:
https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/

Minimum verifiable example:
https://replay.pokemonshowdown.com/gen8customgame-1012540442

Original user report: 
https://www.smogon.com/forums/threads/bug-reports-v3-read-original-post-before-posting.3634749/post-8287732

### PR DESCRIPTION
**Pull request by TZubiri**
Fixes #6027

### CODE REVIEW

File: data/abilities.js [hunk 1]
```diff
@@ -2205,15 +2205,23 @@ let BattleAbilities = {
...
-		desc: "This Pokemon has a random stat raised by 2 stages and another stat lowered by 1 stage at the end of each turn.",
+		desc: "This Pokemon has a random stat raised by 2 stages and another stat lowered by 1 stage at the end of each turn. This does not affect evasion and accuracy",
 		shortDesc: "Raises a random stat by 2 and lowers another stat by 1 at the end of each turn.",
 		onResidualOrder: 26,
 		onResidualSubOrder: 1,
 		onResidual(pokemon) {
 			let stats = [];
 			let boost = {};
-			for (let statPlus in pokemon.boosts) {
-				// @ts-ignore
```
Comment (KrisXV) [inline data/abilities.js]: Why'd you remove this line?

File: data/abilities.js [hunk 2]
```diff
@@ -2205,15 +2205,23 @@ let BattleAbilities = {
...
+			let applicableBoosts = {};
+
+			for (let k in pokemon.boosts){
```
Comment (KrisXV) [inline data/abilities.js]:
```suggestion
			for (let k in pokemon.boosts) {
```

File: data/abilities.js [hunk 3]
```diff
@@ -2205,15 +2205,23 @@ let BattleAbilities = {
...
+				if (k !== 'accuracy' && k!== 'evasion'){
```
Comment (KrisXV) [inline data/abilities.js]:
```suggestion
				if (k !== 'accuracy' && k!== 'evasion') {
```

File: data/abilities.js [hunk 4]
```diff
@@ -2205,15 +2205,23 @@ let BattleAbilities = {
...
+					applicableBoosts[k] = pokemon.boosts[k];
+				}
+			}
+
+			for (let statPlus in applicableBoosts) {
+
```
Comment (KrisXV) [inline data/abilities.js]:
```suggestion
```

File: data/abilities.js [hunk 5]
```diff
@@ -2205,15 +2205,23 @@ let BattleAbilities = {
...
 		onResidual(pokemon) {
 			let stats = [];
 			let boost = {};
-			for (let statPlus in pokemon.boosts) {
-				// @ts-ignore
+			let applicableBoosts = {};
+
+			for (let k in pokemon.boosts){
+				if (k !== 'accuracy' && k!== 'evasion'){
+					applicableBoosts[k] = pokemon.boosts[k];
```
Comment (KrisXV) [inline data/abilities.js]:
```suggestion
					// @ts-ignore
					applicableBoosts[k] = pokemon.boosts[k];
```

File: data/abilities.js (hunk 1 above)
Comment (TZubiri) [reply data/abilities.js]: Blunder

**Comment by Zarel** [comment]
Fixed by 4a3fd8717edafb506affcc

**Pull request closed without merging**